src/Month.jsx

```diff
@@ -252,14 +253,20 @@ let MonthView = React.createClass({
   _headers(row, format, culture) {
     let first = row[0]
     let last = row[row.length - 1]
+    let HeaderComponent = this.props.components.header || Header
 
     return dates.range(first, last, 'day').map((day, idx) =>
       <div
         key={'header_' + idx}
         className='rbc-header'
         style={segStyle(1, 7)}
       >
-        { localizer.format(day, format, culture) }
+        <HeaderComponent
+          date={day}
+          label={localizer.format(day, format, culture)}
+          localizer={localizer}
+          dayFormat={format}
```

Comment:
maybe just `format` for the prop name it may be confusing which format it is across different usages, `weekdayFormat`, `dayFormat`, etc